tests/integration/orcid/test_orcid_push.py

```diff
@@ -0,0 +1,104 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID push tests."""
+
+from __future__ import absolute_import, division, print_function
+
+import mock
+import pkg_resources
+import pytest
+import re
+import requests
+import requests_mock
+import vcr
+
+from redis import StrictRedis
+
+from inspirehep.services.orcid_push.tasks import push_orcid
+
+
+@pytest.fixture
+def mocked_internal_services(api_client):
+    hep_response = api_client.get('/literature/4328')
+    bibtex_response = api_client.get('/literature/4328', headers={
+        'Accept': 'application/x-bibtex',
+    })
+
+    def api_matcher(request):
+        if request.url == 'http://labs.inspirehep.net/api/literature/4328':
+            resp = requests.Response()
+            resp.status_code = 200
+            if request.headers['Accept'] == 'application/json':
+                resp._content = hep_response.data
+                return resp
+            elif request.headers['Accept'] == 'application/x-bibtex':
+                resp._content = bibtex_response.data
+                return resp
+
+    with requests_mock.Mocker() as requests_mocker:
+        requests_mocker.register_uri(
+            requests_mock.ANY,
+            re.compile('.*(indexer|orcid).*'),
+            real_http=True,
+        )
+        requests_mocker.add_matcher(api_matcher)
+        yield requests_mocker
+
+
+@pytest.fixture(scope='function')
+def redis_setup(api):
+    redis_url = api.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+
+    yield r
+
+    r.delete('orcid:0000-0002-2169-2152:4328')
+
+
+@vcr.use_cassette(
+    pkg_resources.resource_filename(
+        __name__,
+        'fixtures/push_casette.yaml'
+    ),
+    decode_compressed_response=True,
+    filter_headers=['Authorization'],
+    ignore_localhost=True,
+)
+def test_push_to_orcid(api, redis_setup, mocked_internal_services):
+    rec_id = 4328
+
+    with mock.patch.dict(api.config, {
+        'ORCID_APP_CREDENTIALS': {
+            'consumer_key': '0000-0002-0970-6486',
+            'consumer_secret': 'secret-redacted',
+        },
+        'ORCID_SANDBOX': True,
+        'SERVER_NAME': 'http://labs.inspirehep.net',
+    }):
+        # Push as new
+        assert not redis_setup.exists('orcid:0000-0002-2169-2152:4328')
+        push_orcid('0000-0002-2169-2152', rec_id, 'token-redacted')
+
+        # Push update
+        assert redis_setup.exists('orcid:0000-0002-2169-2152:4328')
+        push_orcid('0000-0002-2169-2152', rec_id, 'token-redacted')
```

Comment:
Shouldn't you assert something else? How do you know that the update did anything? And that the creation actually created something else than the redis entry?